pkg/asset/agent/manifests/agentclusterinstall.go

```diff
@@ -87,6 +87,23 @@ func (a *AgentClusterInstall) Generate(dependencies asset.Parents) error {
 			serviceNetwork = append(serviceNetwork, cidr.String())
 		}
 
+		machineNetwork := []hiveext.MachineNetworkEntry{}
+		for _, mn := range installConfig.Config.Networking.MachineNetwork {
+			_, cidr, err := net.ParseCIDR(mn.CIDR.String())
+			if err != nil {
+				return errors.Wrap(err, "failed to parse MachineNetwork CIDR")
+			}
+			err = validate.SubnetCIDR(cidr)
+			if err != nil {
+				return errors.Wrap(err, "failed to validate MachineNetwork CIDR")
+			}
```

Comment:
Removed the validate for Machine and Cluster networks. I think we still need the ParseCIDR, no?

Reply:
Actually I think you're right in that the  net.ParseCIDR(xx.CIDR.String()) can be removed since in the install-config the data is in ipnet.IPNet format (e.g. https://github.com/openshift/installer/blob/master/pkg/types/installconfig.go#L356). Its validated by the InstallConfig. Even though the agentclusterinstaller stores the IP addresses as strings they have already been validated and don't need to be done again here.
Updated.